### PR TITLE
Fixed bugs related to library normalization with no_dedup, replicate formatting, and DESeq

### PR DESCRIPTION
**Pull request by epehrsson**
## Changes

Fixes bugs related to library normalization with no_dedup option, replicate formatting, DESeq (single-sample group check and log2FC parameter). Also allows local library installation as workaround for installation issues in shared directory.

## Issues

- fixes https://github.com/CCBR/CARLISLE/issues/126
- fixes https://github.com/CCBR/CARLISLE/issues/123
- fixes https://github.com/CCBR/CARLISLE/issues/119

## PR Checklist

(~Strikethrough~ any points that are not applicable.)

- [x] This comment contains a description of changes with justifications, with any relevant issues linked.
- [x] Update docs if there are any API changes.
- [x] Update `CHANGELOG.md` with a short description of any user-facing changes and reference the PR number. Guidelines: https://keepachangelog.com/en/1.1.0/

### CODE REVIEW

File: workflow/scripts/_diff_markdown.Rmd [hunk 1]
```diff
@@ -51,11 +51,11 @@ pkg_df
...
-if("ELBOW" %in% new.packages){
-  install.packages(paste0(Rlib_dir,"ELBOW_1.10.0.tar.gz"), 
-    repos = NULL, type="source", INSTALL_opts = '--no-lock')
-  new.packages=new.packages[names(new.packages) != "ELBOW"]
-}
+#if("ELBOW" %in% new.packages){
+#  install.packages(paste0(Rlib_dir,"ELBOW_1.10.0.tar.gz"), 
+#    repos = NULL, type="source", INSTALL_opts = '--no-lock')
+#  new.packages=new.packages[names(new.packages) != "ELBOW"]
+#}
```
Comment (kelly-sovacool) [inline workflow/scripts/_diff_markdown.Rmd:58]: Let's go ahead and delete these lines rather than leave them commented-out

```suggestion
```

**Comment by kelly-sovacool** [comment]
@epehrsson can you update the changelog under the `development version` header with a description of these changes?

You can use something like this (or copy-paste this directly)

```
## CARLISLE development version

- Bug fixes (#127, @epehrsson)
    - Removed single-sample group check.
    - Increase memory for DESeq.
    - Ensure control replicate number is an integer.
    - Fix FDR cutoff misassigned to log2FC cutoff.
    - Allow usage of default R library paths.
    - Improve `no_dedup_nreads_genome` variable name.

## CARLISLE v2.5.0
```

**Comment by kelly-sovacool** [comment]
We should update the docs here 

https://github.com/CCBR/CARLISLE/blob/31ecafd7cfe64e97deb3ce46a5bb30f862f09b16/docs/user-guide/preparing-files.md?plain=1#L153-L160

With something like this:

```
**Note**: you must have more than one sample per condition in order to perform differential analysis.
Please ensure each condition contains more than one sample.
```

**Comment by kelly-sovacool** [comment]
@kopardev Maybe now is the time to go ahead and create a container for the DESeq step so we can get rid of the code related to installing R packages? (#128)

**Comment by epehrsson** [comment]
To add more context, the commented-out code allows the user to use their own R library directory instead of the central pipeline directory. Without this edit, rule DESeq fails due to missing packages (within _diff_markdown.Rmd, setup chunk, the CARLISLE_HANDLE_PACKAGES call). In particular, it seems unable to load some ChIPSeeker dependencies. It also no longer handles ELBOW installation. However, the user must install some libraries manually with this edit.

**Comment by kelly-sovacool** [comment]
@epehrsson I chatted with Vishal and he agrees we should switch to using containers. For now, I think the best thing to do would be undo the changes related to the R package installation / lib path code. In a separate PR I'll get rid of that code and add a container.

File: workflow/scripts/_diff_markdown.Rmd [hunk 2]
```diff
@@ -39,7 +39,7 @@ source(params$carlisle_functions)
 Rlib_dir=params$Rlib_dir
 Rpkg_config=params$Rpkg_config
 print(paste0("Using the lib.loc location: ",Rlib_dir))
-assign(".lib.loc", Rlib_dir, envir = environment(.libPaths))
+#assign(".lib.loc", Rlib_dir, envir = environment(.libPaths))
```
Comment (kelly-sovacool) [inline workflow/scripts/_diff_markdown.Rmd:42]:
```suggestion
assign(".lib.loc", Rlib_dir, envir = environment(.libPaths))
```

**Review by kelly-sovacool — APPROVED** [review]

**Pull request merged**